plugins/destination/clickhouse/queries/tables.go

```diff
@@ -5,17 +5,18 @@ import (
 	"slices"
 	"strings"
 
+	"github.com/apache/arrow/go/v17/arrow"
 	"github.com/cloudquery/cloudquery/plugins/destination/clickhouse/client/spec"
 	"github.com/cloudquery/cloudquery/plugins/destination/clickhouse/typeconv/ch/types"
 	"github.com/cloudquery/cloudquery/plugins/destination/clickhouse/util"
 	"github.com/cloudquery/plugin-sdk/v4/glob"
 	"github.com/cloudquery/plugin-sdk/v4/schema"
 )
 
-func sortKeys(table *schema.Table) []string {
+func SortKeys(table *schema.Table) []string {
 	keys := make([]string, 0, len(table.Columns))
 	for _, col := range table.Columns {
-		if col.NotNull || col.PrimaryKey {
+		if (col.NotNull || col.PrimaryKey) && !isCompoundType(col) {
```

Comment:
OK so I was thinking of doing https://github.com/cloudquery/cloudquery/compare/fix/CanonizedField?expand=1

That fixes the underlying issue in `CanonizedField` of losing the nullability and metadata and won't duplicate the logic of https://github.com/cloudquery/cloudquery/blob/a81edd6c5e7c2a25f3396ac80983d28c4af2f1c7/plugins/destination/clickhouse/typeconv/ch/types/nullable.go#L10